libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -335,9 +335,14 @@ void ArrayUtils::copyArrayToMemory(ArrayType const& _sourceType, bool _padToWord
 		if (baseSize > 1)
```

Comment:
I do not understand the code from line 328 to line 333.  Especially why it looks different depending if the source is CALLDATA or MEMORY.  I need to read the callers of this function; (this might be spilling into tomorrow: my estimate was off).